File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/calcudoku/domain/CalcudokuHint.java
package com.zwstudio.logicpuzzlesandroid.puzzles.calcudoku.domain;

public abstract class CalcudokuHint {
    public char op = ' ';
    public int result = 0;
    public String toString() {
        return (result == 0 ? "" : String.valueOf(result)) + String.valueOf(op);
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/calcudoku/domain/CalcudokuHint.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.calcudoku.domain

abstract class CalcudokuHint {
    var op = ' '
    var result = 0
    override fun toString(): String {
        return (if (result == 0) "" else result.toString()) + op.toString()
    }
}